mainpy/__init__.py

```diff
@@ -70,7 +70,10 @@ def callback(self, /) -> Callable[[], _R_co]: ...
     def __call__(self, /) -> _R_co: ...
 
 
-def _is_click_cmd(func: _F) -> TypeGuard[_HasCallbackFunction[_F]]:
+def _is_click_cmd(
+    func: _HasCallbackFunction[object] | _F,
+    /,
+) -> TypeGuard[_HasCallbackFunction[_F]]:
```

Comment:
```suggestion
) -> func is _HasCallbackFunction[_F] if True else False:
```

😃😃😃

Reply:
very based; but won;t that cause issues with other typecheckers?